Create "Add Tooltips to Items" articles (#8911)
DevExpress/devextreme-documentation: includes/tooltip-datagrid-treelist.md

    const dataSource = [ /* ... */ ];
    </script>

##### React

    <!-- tab: App.tsx -->
    import React from 'react';
    import 'devextreme/dist/css/dx.fluent.blue.light.css';
    import { {WidgetName}, Column } from 'devextreme-react/{widget-name}';
    import { Tooltip } from 'devextreme-react/tooltip';

    const dataSource = [ /* ... */ ];

    function DescriptionCell({ data }: { data: { value: string; rowIndex: number } }) {
        return (
            <div>
                <span id={`cell_${data.rowIndex}`}>{data.value}</span>
                <Tooltip
                    target={`#cell_${data.rowIndex}`}
                    showEvent="mouseenter"
                    hideEvent="mouseleave">
                    {data.value}
                </Tooltip>
            </div>
        );
    }

    export default function App() {
        return (
            <{WidgetName} dataSource={dataSource}>
                <Column
                    dataField="description"
                    caption="Description"
                    cellRender={(data) => <DescriptionCell data={data} />}
                />
            </{WidgetName}>
        );
    }

---

> To add a tooltip to a header cell, use the [headerCellTemplate](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/columns/#headerCellTemplate) property with the same approaches described above.

Reviewer: ```suggestion
To add a tooltip to a header cell, use the [headerCellTemplate](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/columns/#headerCellTemplate) property as described above.
```

    import type { CellPreparedEvent } from 'devextreme/ui/{widget-name}';

    const dataSource = [ /* ... */ ];

    export default function App() {
        const onCellPrepared = useCallback((e: CellPreparedEvent) => {
            if (e.rowType === 'data') {
                (e.cellElement as HTMLElement).title = e.value;
            }
        }, []);

        return (
            <{WidgetName} dataSource={dataSource} onCellPrepared={onCellPrepared}>
                <Column dataField="name" caption="Name" />
                <Column dataField="description" caption="Description" />
            </{WidgetName}>
        );
    }

---

To display a styled [Tooltip](/Documentation/ApiReference/UI_Components/dxTooltip/) with custom content, initialize the Tooltip component inside `cellTemplate`:

---
##### jQuery

    <!--JavaScript-->
    $(function() {
        $("#{widgetName}Container").dx{WidgetName}({
            dataSource: dataSource,
            columns: [{
                dataField: "description",
                caption: "Description",
                cellTemplate: function(container, options) {
                    var cellEl = $("<span>")

>                         .attr("id", "cell_" + options.rowIndex)

Reviewer 2: When scrolling virtually in DataGrid, rowIndex recycling can cause duplicate ids
